packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -378,10 +383,13 @@ export function s(
         children = slotset[slotName];
     }
     const vnode = h('slot', data, children);
```

Comment:
The creation of the `slot` VNode is only needed when the component is rendered in the shadow DOM. The code below can be simplified as:

```js
     // ...

     const vmBeingRendered = getVMBeingRendered()!;

     if (!hasShadow(vmBeingRendered)) {
        sc(children);
        return children;
    } else {
        if (vmBeingRendered.renderer.syntheticShadow) {
            sc(children);
        }

        return h('slot', data, children);
    }
```